src/MongoDB.Bson/PreserveAttribute.cs

```diff
@@ -0,0 +1,38 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.ComponentModel;
+
+namespace MongoDB.Bson
```

Comment:
this attribute probably should be placed in some common library for all driver libraries since it has nothing common with Bson logic, but we don't have such place, so I left it MongoDB.Bson

Reply:
Agreed. Since we have no such library placing it here is fine.